src/features/SendAssets/helpers.spec.ts

```diff
@@ -35,78 +35,147 @@ const validERC20ResubmitQuery = {
+    });
+  });
+
+  it('parses valid eth tx query parameters correctly - cancel', () => {
+    const parsedQueryParams = parseQueryParams(validETHCancelQuery)([fNetwork], fAssets, fAccounts);
+    expect(parsedQueryParams).toStrictEqual({
+      type: TxQueryTypes.CANCEL,
+      txConfig: fETHNonWeb3TxConfig
+    });
+  });
+
+  it('fails to derive txConfig when invalid eth tx query parameters are included - cancel', () => {
+    const parsedQueryParams = parseQueryParams(invalidCancelQuery)([fNetwork], fAssets, fAccounts);
+    expect(parsedQueryParams).toStrictEqual({ type: TxQueryTypes.CANCEL, txConfig: undefined });
+  });
+
+  it('fails to derive txConfig when there is no network config for specified chainID - cancel', () => {
+    const parsedQueryParams = parseQueryParams(validETHCancelQuery)([], fAssets, fAccounts);
+    expect(parsedQueryParams).toStrictEqual({ type: TxQueryTypes.CANCEL, txConfig: undefined });
+  });
+
+  it('fails to derive txConfig when there is no added account with from address - cancel', () => {
+    const parsedQueryParams = parseQueryParams(validETHCancelQuery)([fNetwork], fAssets, []);
+    expect(parsedQueryParams).toStrictEqual({ type: TxQueryTypes.CANCEL, txConfig: undefined });
+  });
+
+  it('fails to derive txConfig when invalid eth tx query parameters are included - speed up', () => {
+    const parsedQueryParams = parseQueryParams(invalidSpeedUpQuery)([fNetwork], fAssets, fAccounts);
+    expect(parsedQueryParams).toStrictEqual({ type: TxQueryTypes.SPEEDUP, txConfig: undefined });
   });
 
-  it('fails to derive txConfig when there is no network config for specified chainID', () => {
-    const parsedQueryParams = parseQueryParams(validETHResubmitQuery)([], fAssets, fAccounts);
-    expect(parsedQueryParams).toStrictEqual({ type: 'resubmit', txConfig: undefined });
+  it('fails to derive txConfig when there is no network config for specified chainID - speed up', () => {
+    const parsedQueryParams = parseQueryParams(validETHSpeedUpQuery)([], fAssets, fAccounts);
+    expect(parsedQueryParams).toStrictEqual({ type: TxQueryTypes.SPEEDUP, txConfig: undefined });
   });
 
-  it('fails to derive txConfig when there is no added account with from address', () => {
-    const parsedQueryParams = parseQueryParams(validETHResubmitQuery)([fNetwork], fAssets, []);
-    expect(parsedQueryParams).toStrictEqual({ type: 'resubmit', txConfig: undefined });
+  it('fails to derive txConfig when there is no added account with from address - speed up', () => {
+    const parsedQueryParams = parseQueryParams(validETHSpeedUpQuery)([fNetwork], fAssets, []);
+    expect(parsedQueryParams).toStrictEqual({ type: TxQueryTypes.SPEEDUP, txConfig: undefined });
   });
 });
 
-describe('parseResubmitParams', () => {
-  it('correctly parses valid erc20 resubmit query params', () => {
-    const parsedResubmitParams = parseResubmitParams(validERC20ResubmitQuery)(
+describe('parseTransactionQueryParams', () => {
```

Comment:
No tests for cancel here?

Reply:
`cancel` and `speedup` both use the same transaction query format. They just extend it with `type: 'cancel' | 'speedup'` to tell the interface which banners/errors to display